lib/puppet.rb

```diff
@@ -233,6 +241,9 @@ def self.base_context(settings)
   def self.bootstrap_context
     root_environment = Puppet::Node::Environment.create(:'*root*', [], Puppet::Node::Environment::NO_MANIFEST)
     {
+      :strict => DEFAULT_STRICT_SETTING_VALUE,
+      :tasks => DEFAULT_TASKS_SETTING_VALUE,
+      :code => DEFAULT_CODE_SETTING_VALUE,
```

Comment:
I feel like you may have answered this for me at some point already, but why do these have to go in the bootstrap context and not the base context?

Reply:
off the top of my head I remember that there's some parsing that happens when _loading_ pcore, and to do that correctly there needs to be values for `:strict` and `:tasks`. I might have just added `:code` because it was the next one after that... I'll double check.

Reply:
yeah, I was wrong about the requiring `:code`.